packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -163,7 +167,10 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         getAxisData={this._getAxisData}
         onChartMouseLeave={this._handleChartMouseLeave}
         getDomainMargins={this._getDomainMargins}
-        {...(this._xAxisType !== XAxisTypes.NumericAxis && { xAxisInnerPadding: 2 / 3, xAxisOuterPadding: 0 })}
+        {...(this._xAxisType === XAxisTypes.StringAxis && {
```

Comment:
In the above line, we are only passing props, which are then used as XAxisParams for creating the axis and the padding only works with string axes (band scales).
The bug occurred from the following line in VerticalStackedBarChart where the domainMargin was being calculated for date axis as well.
https://github.com/microsoft/fluentui/blob/fb8782f3294229a18e399bb665453ddcd0b97eab/packages/react-charting/src/components/VerticalStackedBarChart/VerticalStackedBarChart.base.tsx#L1058
But this issue didn't occur in VerticalBarChart.
https://github.com/microsoft/fluentui/blob/fb8782f3294229a18e399bb665453ddcd0b97eab/packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx#L918

Reply:
but I see that you changed the logic from != Numeric to == String on vertical bar chart also

Reply:
Previously, the inner and outer padding props were passed to CartesianChart based on the != Numeric check. These props can be passed in case of numeric or date x-axis as well, but they won't affect the behavior because d3 numeric and date scales don't support padding. I made this change to indicate that these props are only relevant for string x-axis.